Добавлено удаление новостей

StrangeSparrow opened the pull request:
Удаление новостей

StrangeSparrow committed:
Добавлено удаление новостей

StrangeSparrow committed:
addNews

Возможность добавлять новости

StrangeSparrow committed:
Дополнения

StrangeSparrow committed:
Дополнения

Подготовка реализации редактирования

StrangeSparrow committed:
Добавлен редактор

c0e commented on src/main/java/com/mycorp/app/controllers/AdminController.java:
```diff
@@ -0,0 +1,91 @@
+    @Produces(MediaType.TEXT_HTML)
+    @Path("/delete/{id}")
+    public Response deleteNews(@PathParam("id") int id) throws InterruptedException {
+        Thread thread = new Thread(() -> newsService.deleteNews(id));
```
Че-то намудрил с тредом прямо. Сделай просто newsService.deleteNews(id));

c0e replied:
Погугли, может в сервлет че-то добавить надо, чтобы не кешировались страницы

c0e commented on src/main/java/com/mycorp/app/controllers/AdminController.java:
```diff
@@ -0,0 +1,91 @@
+    @Produces(MediaType.TEXT_HTML)
+    @Path("/add")
+    public Response addNews(@FormParam("head") String head, @FormParam("briefly") String briefly, @FormParam("full") String full) throws InterruptedException {
+        Thread thread = new Thread(() -> newsService.addNews(head, briefly, full));
```
Тред тут ни к чему

c0e commented on src/main/java/com/mycorp/app/NewsService.java:
```diff
@@ -5,4 +5,9 @@
 public interface NewsService {
     List<News> fetchNews();
     News fetchSingleNews(int id);
+    void addNews(String head, String briefly, String full);
```
Лучше передавать уже собранный News объект

c0e commented on src/main/java/com/mycorp/app/NewsService.java:
```diff
@@ -5,4 +5,9 @@
+    void addNews(String head, String briefly, String full);
+    void addNews(News news);
+    void deleteNews(int id);
+    void editNews(News news, int id);
```
Давай id уберем в объект News

c0e commented on src/main/java/com/mycorp/app/NewsServiceImpl.java:
```diff
@@ -37,4 +72,30 @@ public News fetchSingleNews(int id) {
+
+    @Override
+    public void fillNews(List<News> newsList) {
+        try (PrintWriter writer = new PrintWriter("webapps/my-app-3.5/WEB-INF/classes/news.csv", "UTF-8")) {
```
в конфиг:  webapps/my-app-3.5/WEB-INF/classes/news.csv

c0e replied:
и для заполнения можно использовать уже готовый addNews, чтобы логика записи была в одном методе

c0e commented on src/main/java/com/mycorp/app/NewsService.java:
```diff
@@ -5,4 +5,9 @@
+    void addNews(News news);
+    void deleteNews(int id);
+    void editNews(News news, int id);
+    public void fillNews(List<News> newsList);
```
это не стоит выносить в интерфейс, т.к. относится исключительно к особенности реализации и в контроллере не используется

c0e commented on src/main/java/com/mycorp/app/controllers/AdminController.java:
```diff
@@ -0,0 +1,55 @@
+    }
+
+    @GET
+    @Path("/page/{id}")
```
Тут надо сделать /news/page/{id} и в остальных методах по аналогии. У тебя же в админке не только новости будут

c0e commented on src/main/java/com/mycorp/app/controllers/AdminController.java:
```diff
@@ -0,0 +1,55 @@
+@Path("/admin")
+public class AdminController {
+    NewsService newsService = new NewsServiceImpl();
+
```
А главной страницей админки можно сделать пустую страницу  с меню: 


<div alighn= "center">
<ul>
<li><a href="/admin/">Главная страница администрирования</a></li>
<li><a href="/admin/news/">Управление новостями</a></li>
</ul>
</div>

StrangeSparrow committed:
test

StrangeSparrow committed:
Исправления

Добавлен файл index.html
Так же добавлены страницы admin.html и news.html, которые отображаются в соответствующих корневых дирректориях. Сделаны некоторые изменения в отображении страниц новостей. Из маппинга убран первоначальный сервлет, который не был создан с помощью Jersey

StrangeSparrow committed:
Ok

StrangeSparrow committed:
Удален тред

StrangeSparrow committed:
Локальный путь убран в конфиг

StrangeSparrow committed:
Исправления
